### ISSUE TITLE
Doesn't play nice with TRAMP

### ISSUE THREAD

**Issue by kevinboulain** [tautologyclub/feebleline#32]
On one Emacs I have `Error running timer ‘feebleline--insert’: (file-error "Process has died")` which completely hides TRAMP's password prompt.
On another it totally make Emacs responsive (or crashes it) :)

I simply run something similar to `emacs /ssh:root@localhost:/`, no special configuration.

**Comment by endian-benjamin** [comment]
Hmm I can't reproduce.

But I used to have `feebleline--insert` wrapped with a `condition-case` that did nothing on error to deal with stuff like that specifically. I figured it was poor practice though so I removed it. But maybe it would be preferable.

**Comment by tautologyclub** [comment]
Are you still experiencing this issue?

**Comment by kevinboulain** [comment]
Woops, sorry, totally forgot this issue :)

Yes, I still have the problem and I even got a repro (beware, it removes `/tmp/trash` to start anew):
```shell
rm -rf /tmp/trash
git init /tmp/tash
cd /tmp/trash

cat > config.el << EOF
(setq user-emacs-directory "/tmp/trash")

(defvar bootstrap-version)
(let ((bootstrap-file
       (expand-file-name "straight/repos/straight.el/bootstrap.el" user-emacs-directory))
      (bootstrap-version 5))
  (unless (file-exists-p bootstrap-file)
    (with-current-buffer
        (url-retrieve-synchronously
         "https://raw.githubusercontent.com/raxod502/straight.el/develop/install.el"
         'silent 'inhibit-cookies)
      (goto-char (point-max))
      (eval-print-last-sexp)))
  (load bootstrap-file nil 'nomessage))
(straight-use-package 'use-package)

(use-package magit
  :straight (:host github :repo "magit/magit"))

(use-package feebleline
  :straight (:host github :repo "tautologyclub/feebleline" :branch "development")
  :config
  (setq feebleline-msg-functions '((magit-get-current-branch :face feebleline-git-face)))
  (feebleline-mode 1))
EOF

# running magit at the same time than tramp will crash
# that's why I did a git init above
emacs -Q -l config.el /ssh:freeshell.org:/
```

You can past the above in a file and execute it with a POSIX shell.

So the problem seems to be the conflict between the processes run by TRAMP and the one run by magit, the latter being started by feebleline.

Note that with this minimal configuration it totally aborts Emacs, with my regular configuration I only get the error in the original post :(

Tested with Emacs 26.1, on macOS and Linux.

**Comment by tautologyclub** [comment]
I really truly can't reproduce: https://ibb.co/bWx6iL

Otoh I couldn't source your script fully either since it got stuck on trying to contact elpa. But as you can see from the scrot, it seems to work totally ok.

**Comment by kevinboulain** [comment]
Actually, it may *seem* to be stuck on ELPA because straight.el will clone the whole repository which is a bit slow due to its size :)

**Comment by kevinboulain** [comment]
A similar issue, talking about using the minibuffer & TRAMP: https://github.com/purcell/emacs.d/issues/88

If you can't reproduce after loading the complete script, feel free to close.
I presume I'll be able to find a workaround in special cases like this one (since commenting out the call to magit fixes the problem in my configuration).

**Comment by tautologyclub** [comment]
I'll try to reproduce again tomorrow - I'll also push a version that doesn't spam minibuffer with error messages when it fails. It feels a bit dirty, but I think it's better than the alternative.

**Comment by kevinboulain** [comment]
For the record, it spams a little when Emacs opens a file in a directory that is later removed:
```
Error running timer ‘feebleline--insert’: (file-missing "Setting current directory" "No such file or directory" "/tmp/a/b/")
```

**Comment by tautologyclub** [comment]
Basically been swamped with work since Nov, so I haven't worked on feebleline. But this is fixed with `cc2a963`
